runtime/src/configs/runtime_params.rs

```diff
@@ -11,28 +11,84 @@ pub mod dynamic_params {
+        /// (i.e. the [`StakeToChallengePeriod`] divided by the [`SpMinDeposit`]).
+        ///
+        /// 300k UNITs / 100 UNITs = 3k ticks (i.e. 5 hours with 6 seconds per tick)
         pub static CheckpointChallengePeriod: BlockNumber = (StakeToChallengePeriod::get()
-            / SpMinDeposit::get()) // 300k UNITs / 100 UNITs = 3k ticks (i.e. 5 hours with 6 seconds per tick)
+            / SpMinDeposit::get())
         .try_into()
         .expect(
             "StakeToChallengePeriod / SpMinDeposit should be a number of ticks that can fit in BlockNumber numerical type",
         );
 
         #[codec(index = 3)]
         #[allow(non_upper_case_globals)]
+        /// 30 ticks, or 3 minutes with 6 seconds per tick.
         pub static MinChallengePeriod: BlockNumber = 30;
+
+        #[codec(index = 4)]
+        #[allow(non_upper_case_globals)]
+        /// Price decreases when system utilisation is below 30%.
+        pub static SystemUtilisationLowerThresholdPercentage: Perbill = Perbill::from_percent(30);
+
+        #[codec(index = 5)]
+        #[allow(non_upper_case_globals)]
+        /// Price increases when system utilisation is above 95%.
+        pub static SystemUtilisationUpperThresholdPercentage: Perbill = Perbill::from_percent(95);
+
+        #[codec(index = 6)]
+        #[allow(non_upper_case_globals)]
+        /// 48 [`PICOUNIT`]s is the price per MB of data, per tick.
+        ///
+        /// With 6 seconds per tick, this means that over a month, the price of 1 GB is:
+        /// 48e-12 * 10 ticks/min * 60 min/h * 24 h/day * 30 days/month * 1024 MB/GB = 21.23e-3 [`UNIT`]s
+        pub static MostlyStablePrice: Balance = 48 * PICOUNIT;
+
+        #[codec(index = 7)]
+        #[allow(non_upper_case_globals)]
+        /// [`MostlyStablePrice`] * 10 = 480 [`PICOUNIT`]s
+        pub static MaxPrice: Balance = MostlyStablePrice::get() * 10;
+
+        #[codec(index = 8)]
+        #[allow(non_upper_case_globals)]
+        /// [`MostlyStablePrice`] / 5 = 9 [`PICOUNIT`]s
+        pub static MinPrice: Balance = MostlyStablePrice::get() / 5;
+
+        #[codec(index = 9)]
+        #[allow(non_upper_case_globals)]
+        /// u = [`UpperExponentFactor`]
+        /// system_utilisation = 1
+        ///
+        /// [`MaxPrice`] = [`MostlyStablePrice`] + u * e ^ ( 1 - [`SystemUtilisationUpperThresholdPercentage`] )
+        ///
+        /// 480 = 48 + u * e ^ (1 - 0.95)
+        /// u = (480 - 48) * e ^ (0.95 - 1) ≈ 411
+        pub static UpperExponentFactor: u32 = 411;
+
+        #[codec(index = 10)]
+        #[allow(non_upper_case_globals)]
+        /// l = [`LowerExponentFactor`]
+        /// system_utilisation = 0
+        ///
+        /// [`MinPrice`] = [`MostlyStablePrice`] - u * e ^ ( [`SystemUtilisationLowerThresholdPercentage`] - 0 )
+        ///
+        /// 9 = 48 - l * e ^ (0.3 - 0)
+        /// l = (48 - 9) * e ^ (-0.3) ≈ 29
+        pub static LowerExponentFactor: u32 = 12;
```

Comment:
Why is this 12 if it should be 29 if we follow the same math as we did for the upper factor and in the docs for this factor?

Reply:
You're absolutely correct, I must have missed it in one of the successive changes I did.